### PR TITLE
Basic code wrote

### PR DESCRIPTION
**Pull request by tarampampam**
| Q             | A
| ------------- | ---
| Bug fix?      | No
| New feature?  | Yes 

## Description

Basic code wrote

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I wrote unit tests for my code
- [ ] I have made changes in [CHANGELOG.md](https://github.com/avto-dev/amqp-rabbit-laravel-queue/blob/master/CHANGELOG.md) file

> About your changes in `CHANGELOG.md`:
>
> * Add new version header like `## v1.x.x`, if it does not exists
> * Add description under `added`/`changed`/`fixed` sections
> * Add reference to closed issues `[#000]`
> * Add link to issue in the end of document

### CODE REVIEW

File: src/Queue.php [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+     * @param int $value
+     *
+     * @return int
+     */
+    protected function normalizePriorityValue(int $value): int
+    {
+        // negative values to zero
+        $value = \max(0, $value);
+
+        // limit max value to 255
```
Comment (Zaboday) [inline src/Queue.php]: Why `255` is not constant?

Comment (tarampampam) [reply src/Queue.php]: Can be oveeriden by user

File: src/Failed/RabbitQueueFailedJobProvider.php [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+     *
+     * @throws Throwable
+     *
+     * @return array|object[]
+     */
+    public function all()
+    {
+        $result = [];
+
+        $this->filterMessagesInQueue($this->queue, function (Message $message) use (&$result): void {
```
Comment (Zaboday) [inline src/Failed/RabbitQueueFailedJobProvider.php:140]: May be better `$this->count() > 0` before start filtering messages?

File: src/Queue.php [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+     * @param string        $queue
+     * @param int|null      $priority Message priority
+     *
+     * @return void
+     */
+    public function push($job, $data = '', $queue = null, ?int $priority = null): void
+    {
+        $options = [];
+
+        if (\is_object($job) && $job instanceof PrioritizedJobInterface) {
```
Comment (Zaboday) [inline src/Queue.php:97]: Same extracting of priority as in `later()`. May be make private method with check `$job` and `$priority`.

File: src/Failed/RabbitQueueFailedJobProvider.php [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+            try {
+                $producer->send($temp_queue, $message);
+            } catch (Exception $e) {
+                $this->exception_handler->report($e);
+            }
+
+            $consumer->acknowledge($message); // anyway
+        }
+
+        unset($consumer);
```
Comment (Zaboday) [inline src/Failed/RabbitQueueFailedJobProvider.php:232]: What purpose of `unset()` variable right before reinitializaton?

**Pull request merged**